### PR TITLE
Edit, Delete Dog, New Dog user roles

### PR DESCRIPTION
**Pull request by ShmulTomer**
## Edit, Delete Dog, New Dog user roles

Issue Number(s): #173 

What does this PR change and why?

Finished,
From my understanding:
Edit Button:
- Managers OR Admins OR Associated Instructors/Caregivers

Delete:
- Managers OR Admins

New Dog button: (previously, instructors/caregivers could see add dog)
- Managers OR Admins

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-fd377ded10b54890b65fd99025488cee?pvs=4) have been updated or are not necessary
- [ ] Relevant reviewers (EM) have been assigned to this PR

### Critical Changes

- Database change / migration to run
- Environment config change
- Breaking API change

### Related PRs

PR Number(s): #number_of_pr

### How To Test

The steps to test the functionality of your ticket. Include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: src/pages/dogs/[id]/index.jsx [hunk 1]
```diff
@@ -466,6 +480,8 @@ export default function IndividualDogPage() {
               ) : (
                 <div className="grow flex gap-4 justify-end">
                   <div className="flex gap-4">
+
+                  {(userRole === "Admin" || userRole === "Manager" || userRole === "Instructor/Caregiver") &&
```
Comment (afazio1) [inline src/pages/dogs/[id]/index.jsx:505]: I don't think there is a role for "Instructor/Caregiver" unless then endpoint was changed. What you can do is check if `dog.association != "Volunteer/Partner"`. If that is the case, we know the user must be a Admin, Manager, or have association Instructor/Caregiver

File: src/pages/dogs/[id]/index.jsx [hunk 2]
```diff
@@ -108,14 +111,25 @@ export default function IndividualDogPage() {
...
+        });
+    }
+  }, [session?.user, status]);
+
   useEffect(() => {
     if (data?.association === "Volunteer/Partner") {
       setShowInfoTab(false);
       setShowLogTab(true);
     } 
+    console.log(data)
```
Comment (afazio1) [inline src/pages/dogs/[id]/index.jsx]: remove

File: src/pages/dogs/index.jsx [hunk 3]
```diff
@@ -55,7 +67,7 @@ export default function DogsPage() {
         setData([]);
       })
       .then((res) => res.json())
-      .then((data) => { setData(data); setLoading(false); } );
+      .then((data) => { console.log(data); setData(data); setLoading(false); } );
```
Comment (afazio1) [inline src/pages/dogs/index.jsx]: remove console.log

**Review by afazio1 — CHANGES_REQUESTED** [review]
Good work just a few questions and small fixes

**Review by laurenmcp — CHANGES_REQUESTED** [review]
Permissions for Instructor/Caregiver look great, but the delete button is still deactivated for Managers/Admin. Thanks for the good work!!

**Comment by ShmulTomer** [comment]
@laurenmcp oops, made the change!

**Review by laurenmcp — APPROVED** [review]
awesome, thanks for the change!

**Comment by afazio1** [comment]
#244
